coherence-pr-review.py

```diff
@@ -0,0 +1,266 @@
+# In the context of topic modeling with LDA, the weights assigned to each word in a topic represent the importance or prevalence of that word within the topic. In the output you provided
+# 
+# Understanding the significance of individual words and their weights within a topic can help provide insights into the key themes and subjects present in the text.
+
+# %%
+def test_personal_context(text):
+    sentiment_scores, actual_topics = evaluate_personal_context(text)
+    print("Sentiment Scores:", sentiment_scores)
+    print()
+    print("Actual Topics:")
+    for topic in actual_topics:
+        print(topic)
+
+# %%
+test_personal_context(data.iloc[0]['Story'])
+
+# %%
+test_personal_context("i lost my best friend and im so happy")
+
+# %% [markdown]
+# the word "my" has the highest weight of 0.102 for the extracted topic.
+# 
+# A high weight for the word "my" suggests that it is a significant term within the topic identified by the LDA model. This means that the word "my" occurs frequently and carries substantial importance within the text when discussing the particular topic associated with that topic index.
+# 
+# In this case, it indicates that personal ownership or possession, likely related to the topic of loss and enduring emotional impact, plays a prominent role in the text. The word "my" may be indicating a personal connection or the speaker's individual perspective in relation to the topic being discussed.
+
+# %% [markdown]
+# ## 3.Sensory details 
+
+# %%
+import nltk
+
+def analyze_sensory_details(text):
+    sentences = nltk.sent_tokenize(text)
+
+    sensory_keywords = {
+        'sight': ['see', 'look', 'watch'],
+        'sound': ['hear', 'listen', 'sound'],
+        'smell': ['smell', 'scent', 'aroma'],
+        'taste': ['taste', 'flavor'],
+        'touch': ['feel', 'touch', 'texture']
+    }
+
+    sensory_details = []
+    for sentence in sentences:
+        lower_sentence = sentence.lower()
+        for sense, keywords in sensory_keywords.items():
+            for keyword in keywords:
+                if keyword in lower_sentence:
+                    sensory_details.append({'sense': sense, 'sentence': sentence})
+                    break
+
+    return sensory_details
+
+
+# %%
+def test_sensory_details(text):
+    sensory_details = analyze_sensory_details(text)
+    print("Sensory Details:")
+    for detail in sensory_details:
+        print(f"{detail['sense']}: {detail['sentence']}")
+
+# %%
+test_sensory_details(data.iloc[2]['Story'])
+
+# %% [markdown]
+# ## 4.Phenominal experience 
+
+# %% [markdown]
+# using emotional analysis 
+
+# %%
+from nltk.sentiment import SentimentIntensityAnalyzer
+
+def analyze_emotional_tone(text):
+    sia = SentimentIntensityAnalyzer()
+    sentiment_scores = sia.polarity_scores(text)
+    return sentiment_scores
+
+speech = "The loss of my father will forever leave an indelible mark on my heart. But im so happy that he died"
+
+emotion_scores = analyze_emotional_tone(speech)
```

Comment:
You get the raw value of scores. But how to convert it to the result? What are the thresholds? I'd encourage you to create another function that will make a decision based on the value of emotional scores.

Reply:
Yes, creating thresholds is one of the things I did in the latest version of the notebook. You can find it in the "metrics" branch.